packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -527,27 +521,27 @@ export class BasicCrawler<
     /**
      * Runs the crawler. Returns a promise that gets resolved once all the requests are processed.
      */
-    async run(): Promise<void> {
-        if (this.isRunningPromise) return this.isRunningPromise;
-
+    async run(): Promise<FinalStatistics> {
         await this._init();
-        this.isRunningPromise = this.autoscaledPool!.run();
         await this.stats.startCapturing();
 
         try {
-            await this.isRunningPromise;
+            await this.autoscaledPool!.run();
```

Comment:
Aren't we gonna miss the promise? I remember it sometimes being useful. But maybe it was an antipattern.